packages/devtools_app/lib/src/code_size/code_size_screen.dart

```diff
@@ -195,22 +308,99 @@ class CodeSizeBodyState extends State<CodeSizeBody>
     );
   }
 
-  Widget _buildTreemap() {
+  Widget _buildTreemap({@required bool showDiff}) {
     return LayoutBuilder(
       key: treemapKey,
       builder: (context, constraints) {
         return Treemap.fromRoot(
-          rootNode: root,
+          rootNode: showDiff ? diffRoot : snapshotRoot,
           levelsVisible: 2,
           isOutermostLevel: true,
           height: constraints.maxHeight,
-          onRootChangedCallback: controller.changeRoot,
+          onRootChangedCallback: showDiff
+              ? controller.changeDiffRoot
+              : controller.changeSnapshotRoot,
         );
       },
     );
   }
 }
 
+class FileImportContainer extends StatelessWidget {
```

Comment:
let's make this a stateful widget that manages the imported file name and then sends the string back upon clicking analyze. You may need to make two widgets - `FileImportContainer` and `DualFileImportContainer` because the "Action" button will be in a different place

Reply:
and you can add a TODO that this widget will eventually manage the file picker.